### PR TITLE
Patch 1.6.1

### PR DESCRIPTION
**Pull request by rishabhatdell**
# Description
CSM-Operator changes for CSM 1.11.1

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/1478 |
|https://github.com/dell/csm/issues/1482|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- Installed csm-operator successfully
- ran all unit tests
- installed csi-unity, csi-powerstore and csi-powerflex using latest samples
- ran cert-csi for all the installed drivers

### CODE REVIEW

**Review by nitesh3108 — APPROVED** [review]

**Review by HarishH-DELL — APPROVED** [review]

**Review by francis-nijay — CHANGES_REQUESTED** [review]
There are multiple yaml lint errors that needs to be fixed

**Comment by francis-nijay** [comment]
We need to make sure all yams that has changes work without errors. Is e2e run succesful?

File: tests/e2e/testfiles/storage_csm_powerscale_resiliency.yaml [hunk 1]
```diff
@@ -293,20 +293,3 @@ spec:
...
-  driver-config-params.yaml: |
-    CSI_LOG_LEVEL: "debug"
-    CSI_LOG_FORMAT: "TEXT"
-    PODMON_CONTROLLER_LOG_LEVEL: "debug"
-    PODMON_CONTROLLER_LOG_FORMAT: "TEXT"
-    PODMON_NODE_LOG_LEVEL: "debug"
-    PODMON_NODE_LOG_FORMAT: "TEXT"
-spec:
-  driver:
-    configVersion: v2.11.0
```
Comment (ChristianAtDell) [inline tests/e2e/testfiles/storage_csm_powerscale_resiliency.yaml:312]: Why are we removing this configmap? What did it do that was wrong? What does removing it unblock?

File: tests/e2e/testfiles/storage_csm_powerscale_observability_top_custom_cert.yaml [hunk 2]
```diff
@@ -1,488 +1,488 @@
-apiVersion: storage.dell.com/v1
```
Comment (ChristianAtDell) [inline tests/e2e/testfiles/storage_csm_powerscale_observability_top_custom_cert.yaml:1]: What happened to this file, why is it being shown as every line removed and re-added? I can't see a difference between the two versions.

File: tests/e2e/testfiles/storage_csm_powerscale_observability.yaml [hunk 3]
```diff
@@ -1,488 +1,488 @@
-apiVersion: storage.dell.com/v1
```
Comment (ChristianAtDell) [inline tests/e2e/testfiles/storage_csm_powerscale_observability.yaml:1]: This is another 500 line difference; what did this change encompass? 

This PR is so large that it is crashing my browser trying to go through the diffs. If all of this is a result of YAML linting, then we should have rolled out those changes incrementally, shouldn't we?

File: bundle/manifests/storage.dell.com_apexconnectivityclients.yaml [hunk 4]
```diff
@@ -12,403 +12,131 @@ spec:
     listKind: ApexConnectivityClientList
     plural: apexconnectivityclients
     shortNames:
-    - acc
+      - acc
     singular: apexconnectivityclient
   scope: Namespaced
   versions:
-  - additionalPrinterColumns:
```
Comment (JacobGros) [inline bundle/manifests/storage.dell.com_apexconnectivityclients.yaml:19]: so, everything in this yaml under line 19 also needed to be brought up a level?

File: bundle/manifests/dell-csm-operator.clusterserviceversion.yaml [hunk 5]
```diff
@@ -1460,1942 +1460,2114 @@ metadata:
...
     repository: https://github.com/dell/csm-operator
     support: Dell Technologies
-  name: dell-csm-operator.v1.6.0
+  name: dell-csm-operator.v1.6.1
   namespace: placeholder
 spec:
   apiservicedefinitions: {}
   customresourcedefinitions:
     owned:
-    - description: ApexConnectivityClient is the Schema for the ApexConnectivityClient
```
[the remark this answers is not in the document]
Comment (JacobGros) [reply bundle/manifests/dell-csm-operator.clusterserviceversion.yaml:1469]: oh I see, the indentation went up a level

File: tests/e2e/testfiles/powerscale-templates/powerscale-auth-secret-template.yaml [hunk 6]
```diff
@@ -1,9 +1,11 @@
 isilonClusters:
   - clusterName: "REPLACE_CLUSTERNAME"
-    username: "REPLACE_USER"
-    password: "REPLACE_PASS"
+    username: "-"
+    password: "-"
     isDefault: true
     endpoint: "REPLACE_AUTH_ENDPOINT"
-    endpointPort: "REPLACE_PORT"
+    endpointPort: REPLACE_AUTH_PORT
```
Comment (ChristianAtDell) [inline tests/e2e/testfiles/powerscale-templates/powerscale-auth-secret-template.yaml:7]: Here we have it as REPLACE_AUTH_PORT with no quotation...

File: tests/e2e/testfiles/powerscale-templates/powerscale-secret-template.yaml [hunk 7]
```diff
@@ -5,4 +5,4 @@ isilonClusters:
     isDefault: true
     endpoint: "REPLACE_ENDPOINT"
     skipCertificateValidation: true
-    endpointPort: 8080
+    endpointPort: "REPLACE_PORT"
```
Comment (ChristianAtDell) [inline tests/e2e/testfiles/powerscale-templates/powerscale-secret-template.yaml:8]: ... and here we have it inside quotation marks. Does it matter that it's in quotes sometimes and not in quotes others?

File: bundle/metadata/annotations.yaml [hunk 8]
```diff
@@ -15,4 +15,4 @@ annotations:
   operators.operatorframework.io.test.config.v1: tests/scorecard/
 
   # Annotations to specify supported OCP versions.
-  com.redhat.openshift.versions: v4.15-v4.16
+  com.redhat.openshift.versions: "v4.15"
```
Comment (JacobGros) [inline bundle/metadata/annotations.yaml:18]: Main branch has:
```
  com.redhat.openshift.versions: "v4.16-v4.17"
 ```
operator 1.6.0 has 
```  com.redhat.openshift.versions: v4.15-v4.16 ```
Are we dropping 4.16 for the patch only? If this is due to yaml fix, we probably want to adopt what main has

File: tests/e2e/testfiles/powerscale-templates/powerscale-auth-secret-template.yaml (hunk 6 above)
Comment (falfaroc) [reply tests/e2e/testfiles/powerscale-templates/powerscale-auth-secret-template.yaml:7]: Thought quotations was causing issues but it doesn't. With or without quotes it works but it was last verified without.

File: tests/e2e/testfiles/powerscale-templates/powerscale-secret-template.yaml (hunk 7 above)
Comment (falfaroc) [reply tests/e2e/testfiles/powerscale-templates/powerscale-secret-template.yaml:8]: It doesn't matter. Tested with and without, both approaches worked but it was last tested with quotes in this scenario.

**Comment by JacobGros** [comment]
Call was made today to not include yaml lint fixes in1.6.1, 
Opened this PR: https://github.com/dell/csm-operator/pull/709 but kept this branch the same in case we ever need to revisit

**Pull request closed without merging**